File: swift/Examples/Hidden/Hidden.swift
import ArgumentParser
import HelpTree

struct Hidden: ParsableCommand {
    static var configuration = CommandConfiguration(
        commandName: "hidden",
        abstract: "Example with hidden commands and flags",
        subcommands: [List.self, Show.self, Admin.self]
    )
    @Option(name: .customLong("debug"), help: .hidden)
    var debug: Bool = false
}

struct List: ParsableCommand {
    static var configuration = CommandConfiguration(commandName: "list", abstract: "List items")
}

struct Show: ParsableCommand {
    static var configuration = CommandConfiguration(commandName: "show", abstract: "Show item details")
    @Argument(help: "Item ID")
    var id: String
}

struct Admin: ParsableCommand {
    static var configuration = CommandConfiguration(
        commandName: "admin",
        abstract: "Administrative commands",
        subcommands: [Users.self, Stats.self, Secret.self]
    )
}

struct Users: ParsableCommand {
    static var configuration = CommandConfiguration(commandName: "users", abstract: "List all users")
}

struct Stats: ParsableCommand {
    static var configuration = CommandConfiguration(commandName: "stats", abstract: "Show system stats")
}

struct Secret: ParsableCommand {
    static var configuration = CommandConfiguration(
        commandName: "secret",
        abstract: "Secret backdoor"
    )
}

@main
struct HiddenEntry {
    static func main() {
        let args = Array(CommandLine.arguments.dropFirst())
        if let invocation = HelpTree.parseInvocation(args) {
            HelpTree.run(for: Hidden.self, invocation: invocation)
        } else {
            Hidden.main()
        }
    }
}

File: rust/crates/help-tree/examples/hidden.rs
use clap::{Args, Parser, Subcommand};
use help_tree::run_with_help_tree;

#[derive(Parser)]
#[command(name = "hidden")]
#[command(about = "Example with hidden commands and flags")]
struct Cli {
    #[arg(long, global = true, help = "Verbose output")]
    verbose: bool,

    #[command(flatten)]
    help_tree: help_tree::HelpTreeArgs,

    #[arg(long, global = true, hide = true, help = "Enable debug mode")]
    debug: bool,

    #[command(subcommand)]
    command: Commands,
}

#[derive(Subcommand)]
enum Commands {
    #[command(about = "List items")]
    List,
    #[command(about = "Show item details")]
    Show {
        #[arg(help = "Item ID")]
        id: String,
    },
    #[command(about = "Administrative commands", hide = true)]
    Admin(AdminArgs),
}

#[derive(Args)]
struct AdminArgs {
    #[command(subcommand)]
    command: AdminCommands,
}

#[derive(Subcommand)]
enum AdminCommands {
    #[command(about = "List all users")]
    Users,
    #[command(about = "Show system stats")]
    Stats,
    #[command(about = "Secret backdoor", hide = true)]
    Secret,
}

fn main() {
    run_with_help_tree::<Cli>(|| {
        let cli = Cli::parse();
        match cli.command {
            Commands::List => println!("List"),
            Commands::Show { .. } => println!("Show"),
            Commands::Admin(_) => println!("Admin"),
        }
    });
}
